subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1487,20 +1488,31 @@ uint8_t ll_adv_enable(uint8_t enable)
 			}
 #endif /* CONFIG_BT_CTLR_ADV_PERIODIC */
 
-			/* Keep aux interval equal or higher than primary PDU
-			 * interval.
+			/* Keep aux interval equal or lower than primary PDU
+			 * interval * (max_skip + 1).
 			 * Use periodic interval units to represent the
 			 * periodic behavior of scheduling of AUX_ADV_IND PDUs
 			 * so that it is grouped with similar interval units
 			 * used for ACL Connections, Periodic Advertising and
 			 * BIG radio events.
 			 */
-			aux->interval =
-				DIV_ROUND_UP(((uint64_t)adv->interval *
-						  ADV_INT_UNIT_US) +
-						 HAL_TICKER_TICKS_TO_US(
-							ULL_ADV_RANDOM_DELAY),
-						 PERIODIC_INT_UNIT_US);
+			if (adv->max_skip == 0U) {
+				/* Special case to keep behaviour unchanged from
+				 * before max_skip was implemented; In this case
+				 * add ULL_ADV_RANDOM_DELAY and round up for a
+				 * aux interval equal or higher instead
+				 */
+				aux->interval =
+					DIV_ROUND_UP(((uint64_t)adv->interval *
+							  ADV_INT_UNIT_US) +
+							 HAL_TICKER_TICKS_TO_US(
+								ULL_ADV_RANDOM_DELAY),
+							 PERIODIC_INT_UNIT_US);
+			} else {
+				aux->interval =
+					(((uint64_t)adv->interval * ADV_INT_UNIT_US) *
+					 (adv->max_skip + 1)) / PERIODIC_INT_UNIT_US;
```

Comment:
Suggest to put `(uint64_t)adv->interval * ADV_INT_UNIT_US)` into a variable to make this more readable. The variable can be used both places. 

Similarly the `/ PERIODIC_INT_UNIT_US` should be possible to have common code for too.